src/ApiException.php

```diff
@@ -138,9 +201,42 @@ public static function createFromRestApiResponse(
             $previous
         );
     }
+    
+    /**
+     * Checks if decoded metadata includes errorInfo message.
+     * If errorInfo is set, it will always contain `reason`, `domain`, and `metadata` keys.
+     * @param array $decodedMetadata
+     * @return array $errorInfo {
+     *     @type boolean $containsErrorInfo
+     *     @type string $reason
+     *     @type string $domain
+     *     @type array $errorInfoMetadata
+     * }
+     */
+    private static function containsErrorInfo(array $decodedMetadata)
+    {
+        $errorInfo = [];
+        $errorInfo['containsErrorInfo'] = false;
+        if (empty($decodedMetadata)) {
+            return $errorInfo;
+        }
+        foreach ($decodedMetadata as $value) {
+            $isErrorInfoArray = isset($value['reason']) && isset($value['domain']) && isset($value['metadata']);
+            if ($isErrorInfoArray) {
+                $errorInfo['containsErrorInfo'] = true;
+                $errorInfo['reason'] = $value['reason'];
+                $errorInfo['domain'] = $value['domain'];
+                $errorInfo['errorInfoMetadata'] = $value['metadata'];
+                return $errorInfo;
```

Comment:
Nit: I might just use a `break` here and let it fall through to the return below.

Reply:
makes sense - done